core/src/processing/core/PGraphics.java

```diff
@@ -7996,7 +7996,25 @@ public final float brightness(int rgb) {
   public int lerpColor(int c1, int c2, float amt) {  // ignore
     return lerpColor(c1, c2, amt, colorMode);
   }
-
+  
+  
+   /**
+   * Generalization of the function lerpColor for many colors that are passed 
+   * by parameter as a vector of colors
+   *
+   * @webref color:creating_reading
+   * @usage web_application
+   * @param c[] interpolate to this set of color
+   * @param amt between 0.0 and 1.0
+   * @see PImage#blendColor(int, int, int)
+   * @see PGraphics#color(float, float, float, float)
+   * @see PApplet#lerp(float, float, float)
+   */
+  int lerpColor(int c[], float amt) {
+    int index = int(c.length * amt);
+    return lerpColor(c[index], c[index + 1 < c.length ? index + 1 : c.length-1], amt * c.length - index);
```

Comment:
Finally, `amt * c.length - index` needs to be changed to `amt * (c.length-1) - index` to match the change made above. Result:
![right](https://cloud.githubusercontent.com/assets/5357642/26529497/b0169708-43b8-11e7-91a5-f835b49aa970.png)